### PR TITLE
Add GCN Tags

### PR DESCRIPTION
**Pull request by mcoughlin**
This PR makes it possible to add GCN tags.

### CODE REVIEW

File: config.yaml.defaults [hunk 1]
```diff
@@ -718,3 +718,12 @@ classifications:
     "variable": "#3a87ad",
     "Time-domain Source": "#ff0000",
   }
+
+gcnTags:
```
Comment (guynir42) [inline config.yaml.defaults:722]: Are these color codes? If so perhaps better to name this config key something more specific?

File: skyportal/handlers/api/gcn.py [hunk 2]
```diff
@@ -199,6 +199,11 @@ def post_gcnevent_from_xml(payload, user_id, session, asynchronous=True):
 
     if skymap is None:
         log(f"No skymap found for event {dateobs}")
+
+        print(tags)
```
Comment (guynir42) [inline skyportal/handlers/api/gcn.py:203]:
```suggestion
```

File: skyportal/handlers/api/gcn.py [hunk 3]
```diff
@@ -462,6 +467,111 @@ async def get(self):
             tags = session.scalars(sa.select(GcnTag.text).distinct()).unique().all()
             return self.success(data=tags)
 
+    @auth_or_token
+    def post(self):
```
Comment (guynir42) [inline skyportal/handlers/api/gcn.py:471]: I'm surprised this works without giving null arguments.

File: static/js/components/GcnTags.jsx [hunk 4]
```diff
@@ -63,22 +107,56 @@ const RenderTags = ({ gcnEvent, show_title = false }) => {
   });
   const localizationTagsUnique = [...new Set(localizationTags)];
 
+  const permission = userProfile.permissions.includes("System admin");
```
Comment (guynir42) [inline static/js/components/GcnTags.jsx:110]: Only the admin gets to delete these? What about users that posted a tag?

**Review by guynir42 — APPROVED** [review]
I have minor comments.

File: config.yaml.defaults (hunk 1 above)
Comment (profjsb) [reply config.yaml.defaults:722]: I agree--this whole section could be called "colors". Ie.

```yaml
colors:
   classifications:
        variable: #xxxx
         ....
    gcn_tags:
         BNS: #xxxx
```

**Comment by profjsb** [comment]
Do you need to also update the classification colorization now that the config section has changed structure?

**Pull request merged**